pkgs/by-name/lr/lrcget/package.nix

```diff
@@ -0,0 +1,99 @@
+{ dbus
+, openssl
+, gtk3
+, webkitgtk
+, pkg-config
+, wrapGAppsHook3
+, fetchFromGitHub
+, buildNpmPackage
+, rustPlatform
+, lib
+, stdenv
+, copyDesktopItems
+, makeDesktopItem
+, alsa-lib
+}:
+let
+  pname = "lrcget";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "tranxuanthang";
+    repo = "lrcget";
+    rev = "${version}";
+    hash = "sha256-9Wtb8Y+cj1ec8vNxvvs3wso62TavNveHRh2Gfx7Br4w=";
+  };
+  frontendBuild = buildNpmPackage {
+    inherit version src;
+    pname = "lrcget-ui";
+    # FIXME: This is a workaround, because we have an git dependency node_modules/lrc-kit contains install scripts
+    # but has no lockfile, which is something that will probably break.
+    forceGitDeps = true;
+    distPhase = "true";
+    dontInstall = true;
+    # To fix `npm ERR! Your cache folder contains root-owned files`
+    makeCacheWritable = true;
+
+    npmDepsHash = "sha256-qQ5UMO3UuD6IvUveTRF35qTlGq5PMbxp1Q4UroDqVtk=";
+
+    # packageJSON = ./package.json;
+    postBuild = ''
+      cp -r dist/ $out
+    '';
+  };
```

Comment:
I think you could add this as its own attribute to the package.

i.e.

```nix
rustPlatform.buildRustPackage rec {
  inherit version src pname;

  frontend = buildNpmPackage {
		...
  };
  ...
}
```
This would also allow you to reference the output of this in `postBuild` using `$frontend`. The main reason for doing this would be to allow overriding this in case someone needs to for whatever reason, as well as allowing users to build the frontend (and test the package more easily)

Reply:
>  The main reason for doing this would be to allow overriding this in case someone needs to for whatever reason, as well as allowing users to build the frontend (and test the package more easily)

if you have time, i would love to see an example for this, ama new in nix